pkg/crc/api/handlers.go

```diff
@@ -65,6 +72,8 @@ func getPullSecretFileContent() (string, error) {
 }
 
 func deleteHandler(_ ArgsType) string {
+	mutex.Lock()
+	defer mutex.Unlock()
 	delConfig := machine.DeleteConfig{Name: constants.DefaultName}
 	r, _ := machine.Delete(delConfig)
```

Comment:
Btw, it since you now have a way to report errors, you could handle the error returned by these commands.

Reply:
So the error inside `r` is handled on the tray side, for the daemon the errors are already logged in the machine package.

Reply:
yeah, more of a design issue with the `machine` package than anything to do with your code :( We really should not hide errors inside results returned by `machine` methods :(